### PR TITLE
Document pod level resources

### PR DESCRIPTION
**Pull request by ndixita**
### Description
~Placeholder~ PR for documentation changes for Pod Level Resources Alpha

### Issue

https://github.com/kubernetes/enhancements/issues/2837

### CODE REVIEW

**Comment by utkarsh-singh1** [comment]
@ndixita, this PR seems to have a lot of commits that seem to have different authors, please only push your changes to the PR.

**Comment by utkarsh-singh1** [comment]
Also add description of the changes you have made on this PR.

**Comment by spurin** [comment]
Hello @ndixita :wave: please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before Tuesday November 19th 2024 18:00 PST. Thank you!

**Comment by spurin** [comment]
Hello @ndixita :wave:, 1.32 Docs Shadow here,

A quick reminder to get your PR ready for review before the deadline, Tuesday November 19th at 18:00 PDT. For additional information, refer to: [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review).

Thank you!

**Comment by jihoon-seo** [comment]
/remove-area blog localization release-eng web-development
/remove-language bn de es fr hi id it ja ko pl pt ru uk vi zh
/remove-sig release

**Comment by sftim** [comment]
If this is reviewable, @ndixita, please update the PR to show that.

**Comment by ndixita** [comment]
working on this today, and will also request an extension

File: content/en/docs/tasks/configure-pod-container/assign-pod-level-resources.md
Comment (haircommander) [inline content/en/docs/tasks/configure-pod-container/assign-pod-level-resources.md:49]: just memory limit?

Comment (haircommander) [reply content/en/docs/tasks/configure-pod-container/assign-pod-level-resources.md:49]: oh I see. I find it weird that the first sentence is resource agnostic "to specify resource requests" vs the limits are specifically about memory

Comment (ndixita) [reply content/en/docs/tasks/configure-pod-container/assign-pod-level-resources.md]: Fixed the sentence, please check if it looks better now

Comment (haircommander) [reply content/en/docs/tasks/configure-pod-container/assign-pod-level-resources.md]: yup looks better! thanks

**Comment by haircommander** [comment]
This is a straightforward example of the work, well done  on the time crunch @ndixita . in the future, I think we could add more detailed docs on the different interactions with OOM score/QoS class, but that's not in scope now I think

LGTM from tech perspective

**Comment by ndixita** [comment]
> This is a straightforward example of the work, well done on the time crunch @ndixita . in the future, I think we could add more detailed docs on the different interactions with OOM score/QoS class, but that's not in scope now I think
> 
> LGTM from tech perspective

I think I missed that we have to write the docs for alpha features. Noting it for next time to add the docs in time.

File: content/en/examples/pods/resource/pod-level-resources.yaml
Comment (ramrodo) [inline content/en/examples/pods/resource/pod-level-resources.yaml:23]:
```suggestion
    resources:
      limits:
        cpu: "0.5"
        memory: "100Mi"
      requests:
        cpu: "0.5"
        memory: "50Mi"
```

**Review by ramrodo — CHANGES_REQUESTED** [review]
There is a typo in the `pod-level-resources.yaml`, in the containers section

File: content/en/examples/pods/resource/pod-level-resources.yaml
Comment (ramrodo) [inline content/en/examples/pods/resource/pod-level-resources.yaml:25]: I tested deploying this Pod but had to delete this second container since having two Nginx containers with the default options seems impossible since both will use the `80` port. Here is the error I faced:

```yaml
$ k -n pod-resources-example logs pod-resources-demo pod-resources-demo-ctr-2
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: Getting the checksum of /etc/nginx/conf.d/default.conf
10-listen-on-ipv6-by-default.sh: info: Enabled listen on IPv6 in /etc/nginx/conf.d/default.conf
/docker-entrypoint.sh: Sourcing /docker-entrypoint.d/15-local-resolvers.envsh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
2024/11/28 03:21:24 [emerg] 1#1: bind() to 0.0.0.0:80 failed (98: Address already in use)
nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
2024/11/28 03:21:24 [emerg] 1#1: bind() to [::]:80 failed (98: Address already in use)
nginx: [emerg] bind() to [::]:80 failed (98: Address already in use)
```

Comment (ndixita) [reply content/en/examples/pods/resource/pod-level-resources.yaml]: changed the image to fedora:latest. PTAL

File: content/en/examples/pods/resource/pod-level-resources.yaml
Comment (sftim) [inline content/en/examples/pods/resource/pod-level-resources.yaml:25]: We prefer not to reference `:latest` images in examples, because the more we do, the more that people infer that this is a good / acceptable practice.

**Review by sftim — COMMENTED** [review]
I'm afraid that this PR doesn't look complete (even for an alpha feature).

I'd expect to see a change to https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/ as well as to the [Tasks](https://kubernetes.io/docs/tasks/) section of the doc (to be clear: the Tasks part is already covered).

For alpha you don't need to write in depth, but the concept explanation of Pod resources does need to cover the option of specifying resource requests and limits at Pod level.

File: content/en/docs/tasks/configure-pod-container/assign-pod-level-resources.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/assign-pod-level-resources.md:57]: Even though this adapts existing content, let's use the official name for this format:
```suggestion
The `args` section in the manifest provides arguments for the container when it starts.
```

**Comment by sftim** [comment]
Here's the key piece of feedback: https://github.com/kubernetes/website/pull/48471#pullrequestreview-2468161934

I think if that gets covered, we can merge the PR and then plan some post-merge fixups.

File: content/en/docs/tasks/configure-pod-container/assign-pod-level-resources.md
Comment (iheartNathan) [inline content/en/docs/tasks/configure-pod-container/assign-pod-level-resources.md:63]: The `namespace` field is included in the manifest already, is there a need to specific it with the `kubectl apply` command?

File: content/en/docs/tasks/configure-pod-container/assign-pod-level-resources.md
Comment (iheartNathan) [inline content/en/docs/tasks/configure-pod-container/assign-pod-level-resources.md:100]: The `namespace` the pod is deployed to is `pod-resources-example`

**Comment by ndixita** [comment]
> This is a straightforward example of the work, well done on the time crunch @ndixita . in the future, I think we could add more detailed docs on the different interactions with OOM score/QoS class, but that's not in scope now I think
> 
> LGTM from tech perspective

@sftim this is where I have approvals from sig-node

**Comment by ndixita** [comment]
> Here's the key piece of feedback: [#48471 (review)](https://github.com/kubernetes/website/pull/48471#pullrequestreview-2468161934)
> 
> I think if that gets covered, we can merge the PR and then plan some post-merge fixups.

Tried to address this. PTAL

File: content/en/docs/tasks/configure-pod-container/assign-pod-level-resources.md
Comment (ndixita) [reply content/en/docs/tasks/configure-pod-container/assign-pod-level-resources.md:74]: It is not needed; although I feel it is okay to still specify namespace in the kubectl for extra clarity. Open to changing it

File: content/en/docs/concepts/configuration/manage-resources-containers.md
Comment (sftim) [inline content/en/docs/concepts/configuration/manage-resources-containers.md:97]: Don't use a `note` callout.
Try this:
```suggestion
## Pod-level resource specification
```

File: content/en/docs/concepts/configuration/manage-resources-containers.md
Comment (sftim) [inline content/en/docs/concepts/configuration/manage-resources-containers.md:115]:
```suggestion
```

File: content/en/docs/concepts/configuration/manage-resources-containers.md
Comment (sftim) [inline content/en/docs/concepts/configuration/manage-resources-containers.md:100]:
```suggestion

{{< feature-state feature_gate_name="PodLevelResources" >}}

```

File: content/en/docs/concepts/configuration/manage-resources-containers.md
Comment (sftim) [inline content/en/docs/concepts/configuration/manage-resources-containers.md:103]:
```suggestion
requires enabling the `PodLevelResources` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/).

With the feature enabled, Kubernetes allows you to declare an
```

File: content/en/docs/concepts/configuration/manage-resources-containers.md
Comment (sftim) [inline content/en/docs/concepts/configuration/manage-resources-containers.md:203]:
```suggestion

{{< feature-state feature_gate_name="PodLevelResources" >}}

```

File: content/en/docs/concepts/configuration/manage-resources-containers.md
Comment (sftim) [inline content/en/docs/concepts/configuration/manage-resources-containers.md:236]: We recommend using `code_example` shortcodes for entire example manifests; see https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#source-code-files (which you are using elsewhere).

File: content/en/docs/reference/command-line-tools-reference/feature-gates/pod-level-resources.md
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/pod-level-resources.md:13]: (nit)
```suggestion
Enable _Pod level resources_:  the ability to specify resource requests and limits at the Pod
level, rather than only for specific containers.
```

File: content/en/docs/tasks/configure-pod-container/assign-pod-level-resources.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/assign-pod-level-resources.md:268]: nit: there are similar lists in the existing pages. Should those other task pages link to this new page?

File: content/en/docs/concepts/configuration/manage-resources-containers.md
Comment (sftim) [inline content/en/docs/concepts/configuration/manage-resources-containers.md:102]: (nit)
```suggestion
the Pod level. At Pod level, Kubernetes {{< skew currentVersion >}} only supports resource requests
or limits for specific resource types: `cpu` and / or `memory`.
This feature is currently in alpha and
```

**Review by sftim — COMMENTED** [review]
Thanks. I've made suggestions that should mean we can merge this if you do another round of edits.

File: content/en/docs/concepts/configuration/manage-resources-containers.md
Comment (ndixita) [reply content/en/docs/concepts/configuration/manage-resources-containers.md:235]: yeah for this file, the other example is not using code_example shortcodes, so for consistency I kept it same. Let me fix it

**Comment by haircommander** [comment]
still LGTM!

**Comment by chanieljdan** [comment]
Docs lgtm & tech lgtm noted

/approve

**Pull request merged**
